### DISCUSSION TITLE
How to determine whether it's a USB mode or Bluetooth mode by using 'FuUdevDevice'

### DISCUSSION THREAD

**Discussion by sam412081go** [fwupd/fwupd#6229] [Q&A]
I have a device that supports both USB mode and Bluetooth mode, and I am currently using 'FuUdevDevice' to implement its Bluetooth OTA functionality. 

In the quirk list, I use` [HIDRAW\VEN_1234&DEV_5678]` to locate the device.
For example for bluetooth mode for quirk file:
```
# My device in BT mode
[HIDRAW\VEN_1234&DEV_5678]
Plugin = my_plugin
GType = FuMyBluetoothDevice
```

but for USB mode, the PID and VID are same as Bluetooth mode. How can I determine which mode it is in and use different GType functions accordingly?

```
# My device in USB mode
[HIDRAW\VEN_1234&DEV_5678]
Plugin = my_plugin
GType = FuMyUSBDevice
```

**Comment by hughsie** [comment]
Good question! Is there a way to query the HID device to know what mode it's in? either looking for something in sysfs or by talking to the device?

**Reply by sam412081go** [reply]
I have two ideas:
1. For usb mode,  use 'FuUsbDevice', and add item `[USB\VID_1234&PID_5678]` in quirk file.  => Need to write a new FuUsbDevice function for USB mode device. Not sure if it's feasible.

2. Like you said, to query the device to know what mode it's in. First, try querying the device using the BT mode in the command prompt. If it's not detected, try querying it again by using  USB method.

**Reply by hughsie** [reply]
> Need to write a new FuUsbDevice function for USB mode device

There's already FuHidDevice that subclasses FuUsbDevice, if that helps.
